04_setup_ironic.sh

```diff
@@ -33,26 +33,59 @@ for IMAGE_VAR in $(env | grep "_LOCAL_IMAGE=" | grep -o "^[^=]*") ; do
-        export $IMAGE_VAR=$LOCAL_REGISTRY_ADDRESS/localimages/${!IMAGE_VAR}
-        sudo podman build -t ${!IMAGE_VAR} .
+        export $IMAGE_VAR=$LOCAL_REGISTRY_ADDRESS:$LOCAL_REGISTRY_PORT/localimages/${!IMAGE_VAR}
+        sudo podman build --authfile $REGISTRY_CREDS -t ${!IMAGE_VAR} .
         cd -
-        sudo podman push --tls-verify=false ${!IMAGE_VAR} ${!IMAGE_VAR}
+        sudo podman push --tls-verify=false --authfile $REGISTRY_CREDS ${!IMAGE_VAR} ${!IMAGE_VAR}
     fi
 
     # Update the bootstrap and master nodes to treat LOCAL_REGISTRY_ADDRESS as insecure
     mkdir -p $OPENSHIFT_INSTALL_PATH/data/data/bootstrap/baremetal/files/etc/containers
-    echo -e "[registries.insecure]\nregistries = ['${LOCAL_REGISTRY_ADDRESS}']" > $OPENSHIFT_INSTALL_PATH/data/data/bootstrap/baremetal/files/etc/containers/registries.conf
+    echo -e "[registries.insecure]\nregistries = ['${LOCAL_REGISTRY_ADDRESS}:${LOCAL_REGISTRY_PORT}']" > $OPENSHIFT_INSTALL_PATH/data/data/bootstrap/baremetal/files/etc/containers/registries.conf
     cp assets/templates/99_local-registry.yaml.optional assets/templates/99_local-registry.yaml
 
     IMAGE_NAME=$(echo ${IMAGE_VAR/_LOCAL_IMAGE} | tr '[:upper:]_' '[:lower:]-')
     OLDIMAGE=$(sudo podman run --rm $OPENSHIFT_RELEASE_IMAGE image $IMAGE_NAME)
     echo "RUN sed -i 's%$OLDIMAGE%${!IMAGE_VAR}%g' /release-manifests/*" >> $DOCKERFILE
 done
 
+if [ ! -z "${MIRROR_IMAGES}" ]; then
+
+    # combine global and local secrets
+    # pull from one registry and push to local one
+    # hence credentials are different
+
+    EXTRACT_DIR=$(mktemp -d "mirror-installer--XXXXXXXXXX")
+    COMBINED_AUTH_FILE=$(mktemp "combined-pullsecret--XXXXXXXXXX")
+    jq -s '.[0] * .[1]' ${REGISTRY_AUTH_FILE} ${REGISTRY_CREDS} | tee ${COMBINED_AUTH_FILE}
+
+    TAG=$( echo $OPENSHIFT_RELEASE_IMAGE | sed -e 's/[[:alnum:]/.]*release://' )
+    MIRROR_LOG_FILE=/tmp/tmp_image_mirror-${TAG}.log
+
+    oc adm release mirror \
+       --insecure=true \
+        -a ${COMBINED_AUTH_FILE}  \
+        --from ${OPENSHIFT_RELEASE_IMAGE} \
+        --to-release-image ${LOCAL_REGISTRY_ADDRESS}:${LOCAL_REGISTRY_PORT}/localimages/local-release-image:${TAG} \
+        --to ${LOCAL_REGISTRY_ADDRESS}:${LOCAL_REGISTRY_PORT}/localimages/local-release-image 2>&1 | tee ${MIRROR_LOG_FILE}
+
+    #To ensure that you use the correct images for the version of OpenShift Container Platform that you selected,
+    #you must extract the installation program from the mirrored content:
+
+    oc adm release extract --registry-config "${COMBINED_AUTH_FILE}" \
+        --command=openshift-baremetal-install --to "${EXTRACT_DIR}" \
+        "${LOCAL_REGISTRY_ADDRESS}:${LOCAL_REGISTRY_PORT}/localimages/local-release-image:${TAG}"
+
+    mv -f "${EXTRACT_DIR}/openshift-baremetal-install" ocp/
+
+    rm -rf "${EXTRACT_DIR}"
+    rm -rf ${COMBINED_AUTH_FILE}
+fi
+
 if [ -f assets/templates/99_local-registry.yaml ] ; then
     build_installer
```

Comment:
No longer needed (It was there do build a custom installer because we had added $OPENSHIFT_INSTALL_PATH/data/data/bootstrap/baremetal/files/etc/containers/registries.conf above)